cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -67,20 +67,21 @@ __global__ void compute_conditional_join_output_size(
     &intermediate_storage[threadIdx.x * device_expression_data.num_intermediates];
 
   std::size_t thread_counter{0};
-  cudf::size_type const start_idx      = threadIdx.x + blockIdx.x * block_size;
-  cudf::size_type const stride         = block_size * gridDim.x;
-  cudf::size_type const left_num_rows  = left_table.num_rows();
-  cudf::size_type const right_num_rows = right_table.num_rows();
-  auto const outer_num_rows            = (swap_tables ? right_num_rows : left_num_rows);
-  auto const inner_num_rows            = (swap_tables ? left_num_rows : right_num_rows);
+  cudf::thread_index_type const start_idx      = threadIdx.x + blockIdx.x * block_size;
```

Comment:
The results of these computations can overflow. You have to cast the rhs values before doing math on them.

Reply:
You can replace this with one line
```
auto const start_idx = cudf::detail::grid_1d::global_thread_id();
```
Reference:
https://github.com/rapidsai/cudf/pull/13968/files#diff-b601b88e04ff05d909c800466b031c6e097fef812300cb2350c8709d6cd37fe2R114

Reply:
Nice thanks David. Done.